### ISSUE TITLE
How to have other users republish mutable content?

### ISSUE THREAD

**Issue by lmatteis** [feross/bittorrent-dht#107]
It seems like nodes are already doing this automatically, through the regular DHT routing stuff, but I need to do this programmatically.

For instance, I publish a signed message using put, and then I leave the network. I want other users to keep it alive by republishing it, and I obviously don't want to provide them with my priv key.

Any help?

**Comment by lmatteis** [comment]
I guess in the `put` users can simply set the `sign` function to return the signature from a previous get right?

```
sign: function (buf) {
    //return ed.sign(buf, keypair.publicKey, keypair.secretKey)
    return <signature returned by get>;
}
```

**Comment by mafintosh** [comment]
You should be able to just put the content you get

``` js
dht.get({k: ...}, function (err, res) {
  // republish this
  dht.put(res)
})
```

This should work for both immutable and mutable content. I'm not sure we have a test for this so if you run into problems let me know.

**Comment by lmatteis** [comment]
Ah very cool. Makes sense. Will try and let you know thanks!

**Comment by pldubouilh** [comment]
@mafintosh I actually thought it was not possible as the nodeID was signed along the rest of the request... I'll see if it works in real life, as it would be an amazing feature !

**Comment by mafintosh** [comment]
@pldubouilh bep44 explicitly allows this in the spec

**Comment by pldubouilh** [comment]
Yep, you're right. For some reason, I thought the signature was stripped on arrival. Very good, so I checked, and I -indeed- receive the signature back, which allow us to republish content.

Although, there's a error when using this

```javascript
dht.get(location, function (err, res) {
  dht.put(res)
})
```
```
 if (!message.a.sig) message.a.sig = opts.sign(encodeSigData(message.a))
TypeError: opts.sign is not a function  at DHT._put
```

... as `DHT.prototype.put` is expecting a function, and not a signature. I just fixed it. I'm running some tests to see if its any good and I'll PR.
